course/content.py

```diff
@@ -1528,7 +1530,14 @@ def get_course_commit_sha(course, participation):
                 try:
                     repo[preview_sha.encode()]
                 except KeyError:
+                    if raise_on_error:
+                        raise CourseCommitShaDoesNotExist(
+                            _("Preview revision '%s' does not exist--"
+                              "showing active course content instead."
+                              % participation.preview_git_commit_sha))
                     preview_sha = None
+                finally:
+                    repo.close()
```

Comment:
Is this necessary if the `repo` is inside a `with` block here?

Reply:
You're right. Removed.